### DISCUSSION TITLE
About Encoder Performance (from email)

### DISCUSSION THREAD

**Discussion by jaehanwork** [seominjoon/kaist-ai605#34] [Q&A] [answered]
Hi professor,

I'm Jaehan, a MS program student at EE.

I have a question about 5.1 of the hw2.

I organized encoder model with the code of Transformer. It showed poor performance (~10 f1 score) comparing to LSTM+Attention+LSTM model (~30 f1 score). Is it normal or did I make some mistake to make the model?

In addition, I embedded the input with concatenation of the context and question. However, it seem that they should be separately embedded and attended by context-question attention for good performance. Could it affect to the performance of the encoder model unlike the LSTM based model?

+ Professor's answer) How many epochs have you trained? It is possible that you haven't trained long enough.

+ Me) Up to 20 epochs, the model was overfitted and early stopped with patience of 10 times. I have tuned the model with several maximum learning rate(from 1e-2 to 1e-5), but it seems that the validation loss cannot decrease. I'll try again. Thanks professor.

**Comment by seominjoon** [comment, marked as answer]
It seems many people had difficulty getting good accuracy because the assignment instruction was not comprehensive enough. Don't worry too much about the accuracy for this assignment. We will rather grade on whether your implementation is in the right direction.

**Reply by jaehanwork** [reply]
Thanks, professor.
